front/src/modules/ui/board/components/BoardOptionsDropdownContent.tsx

```diff
@@ -35,35 +34,36 @@ import { useUpsertView } from '@/ui/view-bar/hooks/useUpsertView';
 import { currentViewScopedSelector } from '@/ui/view-bar/states/selectors/currentViewScopedSelector';
 import { viewsByIdScopedSelector } from '@/ui/view-bar/states/selectors/viewsByIdScopedSelector';
 import { viewEditModeState } from '@/ui/view-bar/states/viewEditModeState';
+import { ViewFieldForVisibility } from '@/ui/view-bar/types/ViewFieldForVisibility';
 
+import { BoardOptionsContext } from '../contexts/BoardOptionsContext';
 import { useBoardCardFields } from '../hooks/useBoardCardFields';
+import { useBoardColumns } from '../hooks/useBoardColumns';
 import { boardCardFieldsScopedState } from '../states/boardCardFieldsScopedState';
 import { boardColumnsState } from '../states/boardColumnsState';
 import { savedBoardCardFieldsFamilyState } from '../states/savedBoardCardFieldsFamilyState';
 import { hiddenBoardCardFieldsScopedSelector } from '../states/selectors/hiddenBoardCardFieldsScopedSelector';
+import { hiddenBoardColumnsSelector } from '../states/selectors/hiddenBoardColumnsSelector';
 import { visibleBoardCardFieldsScopedSelector } from '../states/selectors/visibleBoardCardFieldsScopedSelector';
+import { visibleBoardColumnsSelector } from '../states/selectors/visibleBoardColumnsSelector';
 import { BoardColumnDefinition } from '../types/BoardColumnDefinition';
 import { BoardOptionsDropdownKey } from '../types/BoardOptionsDropdownKey';
 
+import { BoardColumnEditTitleMenu } from './BoardColumnEditTitleMenu';
+
 export type BoardOptionsDropdownContentProps = {
   customHotkeyScope: HotkeyScope;
   onStageAdd?: (boardColumn: BoardColumnDefinition) => void;
 };
 
 type BoardOptionsMenu = 'fields' | 'stage-creation' | 'stages';
 
-type ColumnForCreate = {
-  id: string;
-  colorCode: ThemeColor;
-  index: number;
-  title: string;
-};
-
 export const BoardOptionsDropdownContent = ({
   customHotkeyScope,
   onStageAdd,
 }: BoardOptionsDropdownContentProps) => {
   const { BoardRecoilScopeContext } = useContext(BoardContext);
+  const boardOptionsContext = useContext(BoardOptionsContext);
```

Comment:
I can't seem to deconstruct the properties from context, maybe I am doing something wrong

Reply:
Had to handle the context null check :)